### ISSUE TITLE
Request for hyperparameters to reproduce Mantis results on UCR benchmark

### ISSUE THREAD

**Issue by YXNTU** [ShifengXIE/CauKer#2]
Thanks for the excellent work!

I am using CAUKER to train the Mantis model on the UCR benchmark, but my result is 72.86, which is lower than the reported 78.31 in the paper.

Could you please share the exact hyperparameters or training configurations?

**Comment by ShifengXIE** [comment]
Hi, thanks a lot for your interest in our work and for trying CauKer!

A good way to reproduce the reported result is to first reproduce Mantis using the official training set and default Mantis training pipeline. Then, replace the original training set with the CauKer-generated synthetic data.

For the synthetic data, please make sure to generate more than 100k samples. We recommend using a sequence length of 512 and training for 100 epochs.

Hope this helps, and please feel free to let us know if you still observe a large gap.

**Comment by YXNTU** [comment]
Thanks a lot for the reply.

I generated the synthetic data following README.md (200k samples) using this command:

```bash
python CauKer.py -N 200000 -L 512 -F 4 -P 6 -M 18 -O CauKer200K.arrow
```

For training Mantis, I've been following the instructions in `CauKer/Models/Mantis/readme.md`. 

Could you share the exact training hyperparameters (like learning rate, batch size, etc.) you used for those 100 epochs? Just want to make sure my setup is fully aligned with yours.

**Comment by ShifengXIE** [comment]
Hi, after discussing with the other authors, we suspect you may have run into a test data issue.

In the UCR benchmark, there is a folder containing the processed datasets (around a dozen of them). Please keep those processed versions, and delete the datasets with the same names outside that folder, as those may contain NaN values.

Hope this helps. For CauKer training with Mantis, you can simply run the commands in our repository. We have already verified this setup through multiple rounds of internal and external reproduction.

**Comment by YXNTU** [comment]
I followed your advice and the issue is gone.

Thank you so much for the quick and super helpful response. Seriously, amazing work on CauKer and even better support!
